beacon-chain/blockchain/error.go

```diff
@@ -18,4 +18,28 @@ var (
 	errWrongBlockCount = errors.New("wrong number of blocks or block roots")
 	// block is not a valid optimistic candidate block
 	errNotOptimisticCandidate = errors.New("block is not suitable for optimistic sync")
+	// ErrInvalidBlock is returned when a block is invalid.
+	ErrInvalidBlock = errors.New("invalid block")
 )
+
+type invalidBlock struct {
+	err error
```

Comment:
if you embed the error type you wouldn't even need to implement the Error method, like:
```
type invalidBlock struct {
    error
}
```
but either way is fine!